### ISSUE TITLE
Fascinating - Can I run this locally with Vicuna? and any vector index?

### ISSUE THREAD

**Issue by jasmeetchadha** [bhaskatripathi/pdfGPT#27]
Thanks Bhaskar - can the above be done? Happy to chat more

**Comment by jasmeetchadha** [comment]
So I managed to run it locally but it still uses OPENAI... How can I use a local Vicuna model so that doesnt need OpenAI? Any help would be appreciated. Finally, I would like to use it over all my notes

**Comment by bhaskatripathi** [comment]
That is great !
I think it can be run locally by Vicuna as well. I would find some time to add this feature very soon.

**Comment by bhaskatripathi** [comment]
I would love more contributors to take up these interesting requests.

**Comment by jasmeetchadha** [comment]
One more suggestion would be if this could be run on obsidian markdown notes! Either if I can read all the md files from within a folder or as a plug-in for obsidian. I think your implementation gives very sensible answers

**Comment by bhaskatripathi** [comment]
yes as the code is available under a free license, you can go ahead and add support for open source model without the needs of any credits.

**Comment by jasmeetchadha** [comment]
If only, I knew how to code. Mostly a code-Lego builder

**Comment by bhaskatripathi** [comment]
oh oka no problem. I will create some support for Vicuna and other open sources as I get some bandwidth.

**Comment by kalyanramu** [comment]
Interesting thought, will have to try it when I get time. I tried GPT4All and the performance wasn't great. Perhaps, it might need more prompt engineering. There is a huge demand for open-source models to minimize cost

**Comment by jasmeetchadha** [comment]
Look forward to it
